arbnode/transaction_streamer.go

```diff
@@ -1565,24 +1517,10 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context) er
 
 	if len(pendingTxnsPos) > 0 {
 		fetcher := func(pos arbutil.MessageIndex) ([]byte, error) {
-			msg, err := s.GetMessage(pos)
+			msg, err := getMessageForSubmittingToEspresso(pos, s.GetMessage)
```

Comment:
Move these code to a function in `espresso_utils.go`. This helps writing a unit test for it